SConstruct

```diff
@@ -2471,6 +2471,8 @@ houdiniEnvAppends = {
 		"-std=$CXXSTD",
 	],
 	"CPPFLAGS" : [
+		## \todo: libIECoreHoudini should not use python.
```

Comment:
Quick question : does Houdini itself have a core dependency on Python? If so, what's the harm in IECoreHoudini having one?

Reply:
I believe you can write Houdini executables that don't use python. I'm not sure anyone actually does that, but I think the same could be said of Gaffer, right? To me, its more about following our conventions for code separation. In the context of this PR, I wish I was able to be sure that the "missing to_python by value converter" I'm trying to chase down was not caused by libIECoreHoudini.

Reply:
You can make an executable linking against libGaffer without needing to pull in libpython, yes - the separation is more than just convention. A quick look at Houdini's libs suggests the same isn't the case there though, so I was just wondering if trying to follow convention for IECoreHoudini actually had similar benefits. Not having to worry about `boost::python` at all sounds worth the effort though...

Reply:
I'm surprised to see you're right, 19 of the 21 Houdini libs link to python (and 6 of those to hboost_python). I thought they used to be better about the separation as well, with the aim of having standalone executables using Houdini libs...

Anyways, as you agree, its worth removing the dependency to avoid `boost::python` issues